src/cassette/CassetteImage.cc

```diff
@@ -16,6 +18,26 @@ std::string CassetteImage::getFirstFileTypeAsString() const
 	}
 }
 
+void CassetteImage::setFirstFileType(FileType type, const Filename& fileName) {
+	if (type == UNKNOWN) {
+		// see if there is a hint in the filename
+		auto fileStem = FileOperations::stem(fileName.getResolved());
+		if (fileStem.contains("BLOAD")) {
+			firstFileType = BINARY;
+		} else if (fileStem.contains("CLOAD")) {
+			firstFileType = BASIC;
+		} else if (size_t runPos = fileStem.find("RUN"); runPos != std::string::npos) {
+			if (fileStem.find("CAS", runPos) != std::string::npos) {
+				firstFileType = ASCII;
+			}
```

Comment:
I agree `BLOAD` and `CLOAD` are less likely to accidentally occur as part of the filename. But for reasons of symmetry maybe we should still also check those in a more robust way. Maybe like this? (untested)
```
auto stem = ...;
auto containsInstruction = [&](std::string_view instruction) {
    auto pos = stem.find(instruction);
    if (pos == std::string_view::npos) return false;
    return stem.find("CAS", pos) != std::string_view::npos;
};
if (containsInstruction("BLOAD") {
    firstFileType = BINARY;
} else if (containsInstruction("CLOAD") {
    firstFileType = BASIC;
} else if (containsInstruction("RUN") {
    firstFileType = ASCII;
}
```

Ah, I only notice this now: `std::string_view::npos`, not `std::string::npos`  (in practice this happens to work fine anyway).